401code_challenges/codeChallenge/src/main/java/mergeSort/App.java

```diff
@@ -0,0 +1,64 @@
+package mergeSort;
+
+public class App {
+
+
+    public static void main(String[] args){
+
+        int[] arr= {45,12,9,20};
+        for(int a: arr){
+            System.out.println(a);
+        }
+        System.out.println("==================");
+       int[] res= mergeSort(arr);
+       for(int a: res){
+           System.out.println(a);
+       }
+    }
+
+
+    public static int[] mergeSort(int[] arr){
+
+       return  sortArr(arr,0,arr.length-1);
+
+    }
+
+  private static int[] sortArr(int[] arr,int left,int right){
+        if(left>=right)return  arr;
+
+        int mid=(left+right)/2;
+        sortArr(arr,left,mid);
+        sortArr(arr,mid+1,right);
+      return  merge(arr,left,mid,right);
+
+  }
+
+
+
+  private static int[] merge(int[] arr,int left,int mid,int right){
+      int[] tmp = new int[arr.length];
+      int r1 = mid + 1;
+      int tIndex = left;
+      int cIndex=left;
+
+      while(left <=mid && r1 <= right) {
+          if (arr[left] <= arr[r1])
```

Comment:
Not using brackets on multi-line conditional statements is "dangerous!" It's very easy to cause little bugs that are hard to find